waiter/src/waiter/main.clj

```diff
@@ -69,7 +69,11 @@
                  (not (instance? ManyToManyChannel response))
                  (not (hu/http2? internal-protocol)))
         (try
-          (slurp body)
+          (let [body-stream ^ServletInputStream body
+                buffer (byte-array 1024)]
+            ;; read until the request payload is fully consumed
+            (while (not (.isFinished body-stream))
+              (.read body-stream buffer)))
```

Comment:
Let's add a timer around this body and also get a meter for the number of iterations we run to consume the body.

Reply:
We should also report request bytes streamed like we do at:
https://github.com/twosigma/waiter/blob/b0667031c7356822e792cf9132cb662344ad2fea/waiter/src/waiter/process_request.clj#L320-L336

Reply:
Instead of this loop, we need to read the bytes asynchronously to prevent the Jetty threads from being blocked:
https://github.com/twosigma/waiter/blob/b0667031c7356822e792cf9132cb662344ad2fea/waiter/src/waiter/process_request.clj#L374-L401